src/Query.tsx

```diff
@@ -370,8 +365,16 @@ export default class Query<TData = any, TVariables = OperationVariables> extends
       };
     }
 
-    data.client = this.client;
+    data.client = this.props.client;
 
     return data;
   };
 }
+
+export default class ApolloQuery<TData = any, TVariables = any> extends React.Component<
```

Comment:
We can type `TVariables` as `OperationVariables` instead of any to make it a bit more strict by default.